addons/calendar/static/src/models/activity/activity.js

```diff
@@ -0,0 +1,63 @@
+/** @odoo-module **/
+
+import {
+    registerClassPatchModel,
+    registerInstancePatchModel,
+    registerFieldPatchModel
+} from'@mail/model/model_core';
+import { attr } from '@mail/model/model_field';
+
+registerFieldPatchModel('mail.activity', 'calendar/static/src/models/activity/activity.js', {
+    calendar_event_id: attr({ default: false }),
+});
+
+registerClassPatchModel('mail.activity', 'calendar/static/src/models/activity/activity.js', {
+    /**
+     * @override
+     */
+    convertData(data) {
+        const res = this._super(data);
+        if ('calendar_event_id' in data) {
+            res.calendar_event_id = data.calendar_event_id[0];
+        }
+        return res;
+    },
+});
+
+registerInstancePatchModel('mail.activity', 'calendar/static/src/models/activity/activity.js', {
+    /**
+     * @override
+     */
+    async deleteServerRecord() {
+        if (!this.calendar_event_id){
+            await this._super();
+        } else {
+            await this.async(() => this.env.services.rpc({
+                model: 'mail.activity',
+                method: 'unlink_w_meeting',
+                args: [[this.id]],
+            }));
+            this.delete();
+        }
+    },
+
+    /**
+     * In case the activity is linked to a meeting, we want to open the calendar view instead.
+     *
+     * @override
+     */
+    async edit() {
+        if (!this.calendar_event_id){
+            this._super();
+        } else {
+            const action = await this.async(() => this.env.services.rpc({
+                model: 'mail.activity',
+                method: 'action_create_calendar_event',
+                args: [[this.id]],
+            }));
+            this.env.bus.trigger('do-action', {
+                action
+            });
+        }
+    }
+});
```

Comment:
Would have been best to make another method (e.g. `reschedule()`) than override `edit()`, especially when the change in `Activity` component consists of defining a new button `reschedule`.

If instead you wanted to define reschedule as an edit for activities that are linked to calendar event, then you should customize edit button instead of duplicating the button with only a change of icon and label.

IMO I prefer defining "reschedule" as a new feature, so I would pick the former approach.

Just wanted to point out that implementation in this PR is very confusing because it's a mix of both approaches, so it's edit but not really edit because it's schedule, but it behaves like an edit and is handled as an edit... but actually we create a calendar event... because this is reschedule and not an edit... Right?

Reply:
The initial thinking probably comes from the previous implementation of this feature where the meeting specific behavior is also implemented within the edit function:
https://github.com/odoo/odoo/blob/14.0/addons/calendar/static/src/js/mail_activity.js#L16

Actually the functional behavior is also a bit strange because indeed it's called "reschedule", but it allows you to edit the activity by editing the linked meeting, or move the linked meeting to move the activity to another date and time.

But now that I re-read the code, I do agree with you, would have been better to clearly separate the 2 buttons / behaviors.